clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -367,6 +367,8 @@ def err_drv_fsycl_with_pch : Error<
   "Precompiled header generation is not supported with '-fsycl'">;
 def err_drv_fsycl_unsupported_with_opt
   : Error<"'%0' is not supported with '-fsycl'">;
+def warn_drv_fsycl_host_compiler_options
+  : Warning<"'%0' should be used only in conjunction with '-fsycl-host-compiler'">, InGroup<InvalidCommandLineArgument>;
```

Comment:
I think this would be better as a more generic warning:  `warn_drv_opt_requires_opt` and is associated with `InGroup<UnusedCommandLineArgument>` - thoughts?

Reply:
Since `-fsycl-host-compiler` is not passed in the command line, there's no way to get this value from Arg, so I had to include '-fsycl-host-compiler' in the warning text and hence the name is specific to -fsycl-host-compiler.
Also, since this warning is about needing to use `-fsycl-host-compiler-options` with -`fsycl-host-compiler`, I felt `InvalidCommandLineArgument` is more appropriate than `UnusedCommandLineArgument`
I am okay if you prefer `UnusedCommandLineArgument`

Reply:
True, there is no passed in arg associated, but you can pass in `"-fsycl-host-compiler"` as a string instead.  It just gives us flexibility on re-using the diagnostic later on.  I see other 'requires' type options in the diagnostic grouping that use `UnusedCommandLineArgument` which is why I suggested it.